storm-core/src/jvm/backtype/storm/scheduler/SupervisorDetails.java

```diff
@@ -119,10 +119,18 @@ private Double getTotalResource(String type) {
     }
 
     public Double getTotalMemory() {
-        return getTotalResource(Config.SUPERVISOR_MEMORY_CAPACITY_MB);
+        Double totalMemory = getTotalResource(Config.SUPERVISOR_MEMORY_CAPACITY_MB);
+        if (totalMemory == null) {
+            throw new IllegalStateException("default value for supervisor.memory.capacity.mb is not set!");
+        }
+        return totalMemory;
     }
 
     public Double getTotalCPU() {
-        return getTotalResource(Config.SUPERVISOR_CPU_CAPACITY);
+        Double totalCPU = getTotalResource(Config.SUPERVISOR_CPU_CAPACITY);
+        if (totalCPU == null) {
+            throw new IllegalStateException("default value for supervisor.cpu.capacity is not set!");
+        }
+        return totalCPU;
```

Comment:
This case should not really happen.  This will only happen if the config SUPERVISOR_CPU_CAPACITY was not included in defaults.yaml for some reason